### ISSUE TITLE
statsd to Opentelemetry conversion issue

### ISSUE THREAD

**Issue by povilasv** [influxdata/telegraf#12507]
### Relevant telegraf.conf

```toml
[global_tags]
  env = "stage"
[agent]
  interval = "5s"
  round_interval = true
  metric_batch_size = 1000
  metric_buffer_limit = 10000
  collection_jitter = "0s"
  flush_interval = "10s"
  flush_jitter = "5s"
  precision = ""
  debug = true
  quiet = false
  logfile = "/var/log/telegraf/telegraf.log"
  logfile_rotation_max_size = "5000MB"
  logfile_rotation_max_archives = 2
  log_with_timezone = "local"
  omit_hostname = true
[[inputs.statsd]]
   protocol = "udp"
   max_tcp_connections = 250
   tcp_keep_alive = false
   service_address = ":8080"
   delete_gauges = true
   delete_counters = true
   delete_sets = true
   delete_timings = true
   percentiles = [90.0, 95.0, 99.0]
   metric_separator = "."
   parse_data_dog_tags = false
   allowed_pending_messages = 100000
   percentile_limit = 100
[[outputs.opentelemetry]]
    service_address = "endpont-here"
    compression = "gzip"
```
```


### Logs from Telegraf

```text
unrelated
```

### System info

1.25.0

### Steps to reproduce

1. Run this app https://github.com/devlibx/gox-metrics
2. Setup Telegraf
3. Send metrics to OpenTelemetry backend
...

### Expected behavior

When `delete_counters= true`, the OpenTelemetry Counter should be sent with Delta Temporality. As Telegraf accumulates data for every interval and then resets counters. Example we send `200` every interval, we see telegraf outputs following:

T0: 200
T1: 200
T2: 200

Which is Delta Temporality, more info -> https://opentelemetry.io/docs/reference/specification/metrics/supplementary-guidelines/#aggregation-temporality

When `delete_counters= false`, the OpenTelemetry Counter should be sent with Cumulative Temporality. As Telegraf accumulates the data and reports cumulative counter. For example:

T0: 200
T1: 400
T2: 600

### Actual behavior

We always send Temporality as Cumulative in this code:

 https://github.com/influxdata/influxdb-observability/blob/28e34bc69b46ddaab673aa784e67e467f08902e8/influx2otel/metrics.go#L158-L161

**Comment by jacobmarble** [comment]
I think this is a good idea. As I understand (please correct me):

The OpenTelemetry metrics data model for Sum, Histogram, and ExponentialHistogram [requires an "aggregation temporality" attribute](https://opentelemetry.io/docs/reference/specification/metrics/data-model/#temporality).

The Telegraf statsd input plugin more-or-less coerces the statsd data model to the Prometheus data model. Both statsd and Prometheus don't signal "aggregation temporality", but statsd is compatible with "delta" and Prometheus is compatible with "cumulative".

The Telegraf statsd input plugin has option `delete_counters` to signal whether Telegraf should accumulate (making compatible with Prometheus), or pass through, sum and histogram values.

The `influx2otel` metrics conversion doesn't have a way to detect aggregation temporality, in fact, it is hard coded to signal cumulative, because it assumes Prometheus data model.

Certain line protocol inputs, such as statsd-via-Telegraf, are capable of generating "delta aggregation temporality", so we should add support somehow, without breaking too many things.

---

How to fix?

- I agree that we need a line protocol tag to signal aggregation temporality, and not simply assume that all metrics are cumulative.
- Because this new tag specifically satisfies a concern in the OpenTelemetry data model, it should have a name such as `otel.aggregation_temporality`, with values `delta` and `cumulative`.
    - When the new tag is missing, assume cumulative (as is the behavior now).
- If the accumulation lives in Telegraf (the case in the statsd input plugin when `delete_counters = false`, correct?) then we should add the start time to the line protocol data model as field `otel.start_time_unix_nano`. [In a related issue](https://github.com/influxdata/telegraf/issues/12664), I indicated that a metric start time should correlate with the time the accumulation began, which is possible in this case, because the statsd plugin is accumulating.

For context, `otel2influx` drops delta metrics in the Prometheus output schemas. That could be similarly adjusted, separately.

---

@povilasv please share your thoughts on the above.

**Comment by Hipska** [comment]
Also know you can filter the tags you don't want on an output config.

**Comment by powersj** [comment]
> And this creates a huge cardinality issue in Prometheus which will eventually make Prometheus OOM.

I did not realize this was an issue for Prometheus similar to influx

> So what we basically want is somehow pass the start_time, but make it in a way that's invisible to the output plugins that don't need that.

Instead of at output plugins, is this something that could be handled at the parser level? If the parser see's the tag that Jacob suggested it goes looking for the corresponding start date field.

**Comment by powersj** [comment]
Step 1: get the data - as @jacobmarble mentioned let's add a config option to the statsd option that will set a field (`otel.start_time_unix_nano`) and tag (`otel.aggregation_temporality`). This is a modification to your PR in #13087.

This gives us an opt-in method of adding this data to avoid breaking existing users and provides us a way to collect it and know if we have it.

Step 2: Now that you have the data, update the opentelementy output and/or the prometehus serializer with a config option that when set, looks at a specific tag to set the start time.

I think this is what @jacobmarble was suggesting in his previous comment.

**Comment by powersj** [comment]
@povilasv it looks like the BSD fix was released a couple days ago in v1.16.0/v0.39.0. What needs to happen next?
